agent/src/agentclient.rs

```diff
@@ -65,9 +65,11 @@ impl<T: APIServerClient> BrupopAgent<T> {
     }
 
     #[instrument(skip(self), err)]
-    pub async fn check_node_custom_resource_exists(&mut self) -> Result<bool> {
-        // try to check if node custom resource exist in the store first. If it's not present in the
-        // store(either node custom resource doesn't exist or store data delays), make the API call for second round check.
+    async fn check_node_shadow_exists(
+        &self,
+    ) -> std::result::Result<bool, agentclient_error::BottlerocketShadowRWError> {
```

Comment:
I was struggling to return agentclient_error::Error but somehow it's very hard to implement that. Since this is a private method and the Err will later be turned into agentclient_error::Error, I would assume it is okay to just return agentclient_error::BottlerocketShadowRWError here.